tools/IncrementVersion.ps1

```diff
@@ -1,21 +1,55 @@
-$versionPath = "$PSScriptRoot\CustomMSBuild\Versioning.props"
+<#
+.PARAMETER versionPath
+  The path to the msbuild props file where the version number is specified
+.PARAMETER lastReleaseCommit
+  The ID of the last commit to be released
+.PARAMETER forceMinorIncrement
+  Whether to force an increment of the minor version number, regardless of API changes since the last release
+.PARAMETER forceRevisionIncrement
+  Whether to force an increment of the revision version number, regardless of API changes since the last release
+#>
+
+Param(
+  [string]
+  $versionPath,
+  [string]
+  $lastReleaseCommit,
+  [Switch]
+  $forceMinorIncrement,
+  [Switch]
+  $forceRevisionIncrement
+)
+
+if ($versionPath -eq "")
+{
+  $versionPath = "$PSScriptRoot\CustomMSBuild\Versioning.props"
+}
+
+if ($lastReleaseCommit -eq "")
+{
+  # use git to find the commit ID of the last change to the versioning.props file; we are assuming that this file is only changed to increment version numbers during release
+  $lastReleaseCommit = git log -n 1 --pretty=format:%H -- $versionPath
+}
 
-$lastReleaseCommit = git log -n 1 --pretty=format:%H -- $versionPath
 Write-Host "The last release's commit was $lastReleaseCommit"
 
+# use git to find the list of file paths that have been changed since the last commit
 $changedFiles = git diff --name-only HEAD $lastReleaseCommit
 
 $apiChanges = $false
+$breaks = @()
+# for each changed file
 ForEach ($line in $($changedFiles -split "`r`n"))
```

Comment:
Is "`r`n" portable?

Reply:
Probably not, switched to `Environment.NewLine` instead